packages/components/src/confirm-dialog/component.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import { forwardRef, useEffect, useState } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import type { Ref, SyntheticEvent, KeyboardEvent } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Modal from '../modal';
+import type { OwnProps } from './types';
+import {
+	useContextSystem,
+	contextConnect,
+	WordPressComponentProps,
+} from '../ui/context';
+import { Flex } from '../flex';
+import Button from '../button';
+import * as styles from './styles';
+import { useCx } from '../utils/hooks/use-cx';
+
+function ConfirmDialog(
+	props: WordPressComponentProps< OwnProps, 'div', false >,
+	forwardedRef: Ref< any >
+) {
+	const {
+		isOpen,
+		message,
+		onConfirm,
+		onCancel = () => {},
+		...otherProps
+	} = useContextSystem( props, 'ConfirmDialog' );
+
+	const cx = useCx();
+	const wrapperClassName = cx( styles.wrapper );
+
+	const [ _isOpen, setIsOpen ] = useState< Boolean >();
+	const [ _selfClose, setSelfClose ] = useState< Boolean >();
+
+	useEffect( () => {
+		// We only allow the dialog to close itself if `isOpen` is *not* set.
+		// If `isOpen` is set, then it (probably) means it's controlled by a
+		// parent component. In that case, `selfClose` might do more harm than
+		// good, so we disable it.
+
+		const isIsOpenSet = typeof isOpen === 'boolean';
```

Comment:
Nit, but in my mind more clear

```suggestion
		const isIsOpenSet = typeof isOpen !== 'undefined';
```

Reply:
👍🏻